benchmark/src/ttft/index.ts

```diff
@@ -0,0 +1,97 @@
+import cp from "node:child_process";
+import fs from "node:fs";
+import path from "node:path";
+import { TURBO_BIN, type TTFTData } from "../helpers";
+import { run } from "./run";
+
+const profileFile = process.argv[2]; // Should be "windows", "ubuntu" or "macos"
+
+if (!profileFile) {
+  console.error("Error: Missing profile name");
+  printUsageMessage();
+  process.exit(1);
+}
+
+if (!profileFile.endsWith(".json")) {
+  console.error("Error: please provide a profile name ending in .json");
+  printUsageMessage();
+  process.exit(1);
+}
+
+const profileName = path.basename(profileFile, ".json");
+
+const profileExt = path.extname(profileFile); // Should always be .json, but we'll get it from here anyway.
+
+const ttftFileName = `${profileName}-ttft${profileExt}`;
+
+// process.cwd() should be benchmark/ here.
+const fullProfilePath = path.join(process.cwd(), profileFile);
+const ttftFilePath = path.join(process.cwd(), ttftFileName);
+
+console.log(`Profile will be saved to ${fullProfilePath}`);
+console.log(`TTFT data will be saved to ${ttftFilePath}`);
+
+if (fs.existsSync(fullProfilePath)) {
+  console.error(`Error: ${fullProfilePath} already exists`);
+  printUsageMessage();
+  process.exit(1);
+}
+
+if (!fs.existsSync(TURBO_BIN)) {
+  throw new Error("No turbo binary found");
+}
+
+cp.execSync(`${TURBO_BIN} --version`, { stdio: "inherit" });
+
+run(fullProfilePath); // Actual benchmark
+
+const profileJSON = JSON.parse(fs.readFileSync(fullProfilePath).toString());
```

Comment:
We need to define the schema of this JSON as a Typscript interface and that should clear up all of the TS errors.
```suggestion
const profileJSON = JSON.parse(fs.readFileSync(fullProfilePath).toString()) as ProfileJson;
```

Reply:
i could do that, didn't seem super useful tbh and each item in profile.json can look different i think, especially `args`